content/docs/command-reference/params/diff.md

```diff
@@ -19,35 +19,29 @@ positional arguments:
 
 ## Description
 
-This command provides a quick way to compare parameter values among experiments
-in the repository history. Requires that Git is being used to version the
-project params.
+Provides a quick way to compare parameter values among experiments in the
+repository history. Requires that Git is being used to version the project
+params.
 
-> Parameter dependencies are defined with the `-p` option in `dvc run`. See also
-> `dvc params`.
+> Parameter dependencies are defined in the `params` field of `dvc.yaml` (e.g.
+> with the the `-p` (`--params`) option of `dvc run`).
 
 Without arguments, this command compares parameters currently present in the
 <abbr>workspace</abbr> (uncommitted changes) with the latest committed version.
-
-Supported parameter _value_ types are: string, integer, float, and arrays. DVC
-itself does not ascribe any specific meaning for these values.
-
-❗ By default it only shows parameters that were changed.
+This includes everything in the default parameters file (`params.yaml`) as well
+as all `params` found in [DVC files](/doc/user-guide/dvc-files). Only params
+that have changes are listed.
 
 ## Options
 
-- `--targets <paths>` - limit command scope to these params files. When
-  specifying arguments for `--targets` before `revisions`, you should use `--`
-  after this option's arguments, e.g.:
+- `--targets <paths>` - limit or expand the command's scope to these params
+  files.
 
-  ```dvc
```

Comment:
> dvc params diff can actually operate on any file now

On that I did the change aboveto the `--targets` option. Is that what you meant? I should probably test this because I'm not sure it makes sense: if the param is was never registered in dvc.lock can it still diff it to previous versions? Will check...

Reply:
I tested this and found some possible bugs: iterative/dvc/issues/5186 so I reverted the part about "or expand" for now. Will review the text again once I have an answer and better understand `params diff --targets` which as of now doesn't seem to do much...

Reply:
> if the param is was never registered in dvc.lock can it still diff it vs. previous versions? Will check... ⌛

it exactly how it works. You don't have to register anything with `dvc.yaml` now for params and metrics. You don't even need it to be a DVC project.

Reply:
>  You don't have to register anything with dvc.yaml now for params

Gotcha, so it's purely based of Git commits. OK, updated! PTAL.

Reply:
> and metrics

Updated that ref. too. What about `plots diff`?

Reply:
> What about plots diff?

Most likely not, it should be a ticket on GH to my mind to cover that.

Reply:
I confirmed that I also applies to plots. All updated!

Reply:
where does it take Vega templates if DVC is not initialized?

Reply:
> if DVC is not initialized

Right, I didn't test without DVC, and I didn't mention that either. But it works for plots files that dvc.yaml isn't tracking — that's what I meant.

Reply:
> I didn't mention that either in any ref.

Should I? It seems like an edge case/ side-effect to me. Not sure it's worth documenting but maybe. Something like "this command doesn't require an existing DVC project to run in." (from [get](https://dvc.org/doc/command-reference/get)).

Reply:
> should be a ticket on GH to my mind to cover that

BTW yes there's iterative/dvc/issues/4446 so we can also decide on what to doc as that issue gets' implemented and docs PRs are created for those changes.

Reply:
> It seems like an edge case/ side-effect to me. 

it's not side effect - it's intentionally made that you don't need DVC project if you want capturing metrics, params, plots

Reply:
Okdk. Added a note to metrics diff and params diff about this (no DVC project required when using --targets, only a Git repo).